### PR TITLE
Extend `XPENDING` and add support for `XGROUP {CREATECONSUMER,DESTROY,DELCONSUMER}`, `XINFO {CONSUMERS,GROUPS}`, `XCLAIM`

### PR DESCRIPTION
**Pull request by sandyharvie**
As stated in the title, this PR adds support for the following commands:
- `XGROUP {CREATECONSUMER,DESTROY,DELCONSUMER}`
-  `XINFO {CONSUMERS,GROUPS}`
-  `XCLAIM`

In addition, I've added support for `IDLE` to the `XPENDING` implementation.

### CODE REVIEW

File: cmd_stream.go [hunk 1]
```diff
@@ -300,19 +301,37 @@ func (m *Miniredis) makeCmdXrange(reverse bool) server.Cmd {
...
+			subCmd,
+		))
 	}
 }
 
 // XGROUP CREATE
 func (m *Miniredis) cmdXgroupCreate(c *server.Peer, cmd string, args []string) {
-	stream, group, id := args[1], args[2], args[3]
+	if len(args) != 3 && len(args) != 4 {
+		c.WriteError(errWrongNumber("CREATE"))
```
Comment (alicebob) [inline cmd_stream.go:331]: All errors also need a `setDirty(c)` (for transactions).

**Comment by alicebob** [comment]
Hi! Thanks for the PR. It looks good on first glance, but I'll need some time to have a proper look.

File: cmd_stream.go (hunk 1 above)
Comment (sandyharvie) [reply cmd_stream.go:332]: Got it–fixed!

**Comment by alicebob** [comment]
PR looks good. I'm still trying to fix the problem with pre 1.17 Go, but I might also say the minimum version of miniredis will be 1.17 (which seems reasonable to me since it's the last version pre-generics).

File: cmd_stream.go [hunk 2]
```diff
@@ -311,8 +317,8 @@ func (m *Miniredis) cmdXgroup(c *server.Peer, cmd string, args []string) {
 		m.cmdXgroupCreateconsumer(c, cmd, args)
 	case "DELCONSUMER":
 		m.cmdXgroupDelconsumer(c, cmd, args)
-	case "HELP":
-	case "SETID":
+	case "HELP",
+		"SETID":
```
Comment (sandyharvie) [inline cmd_stream.go:321]: Ah, good catch–thanks!

**Comment by sandyharvie** [comment]
> PR looks good. I'm still trying to fix the problem with pre 1.17 Go, but I might also say the minimum version of miniredis will be 1.17 (which seems reasonable to me since it's the last version pre-generics).

I think I've fixed the remaining issues in https://github.com/alicebob/miniredis/pull/281/commits/65aebb7f823482339cf846fa7bcc798f44f27e85, but feel free to revert the changes if you choose to make 1.17 the minimum supported version.

**Comment by alicebob** [comment]
Now CI is happy, thanks. Can you squash all the commits? I'll have a final look then (it's a big pr :)

**Comment by alicebob** [comment]
thanks!

**Pull request merged**